### PR TITLE
Ensure Rails module has #env

### PR DESCRIPTION
**Pull request by eigilsagafos**
This addresses #3402

### CODE REVIEW

File: lib/sidekiq/testing.rb [hunk 1]
```diff
@@ -317,7 +317,7 @@ def drain_all
   end
 end
 
-if defined?(::Rails) && !Rails.env.test?
+if defined?(::Rails) && Rails.respond_to?(:env) && !Rails.env.test?
```
Comment (seuros) [inline lib/sidekiq/testing.rb]: if defined?(::Rails::VERSION)

**Comment by eprothro** [comment]
@mperham personally, I'd love to see a release for this fixup, I would wager this breaks more than eigilsagafos and our gems dependent on sidekiq but not rails.

**Comment by mperham** [comment]
I don't like to issue quick patch releases unless a large percentage of users are affected.  You can put this in your initializer to work around:

```ruby
# Remove once running Sidekiq 4.2.11+
module Rails
  def self.env
    self
  end
  def self.test?
    true
  end
end
```

**Comment by eprothro** [comment]
Yep, already have. Thanks, figured I would check.

**Pull request merged**